packages/core/src/lib/implementation/execute-plugin.ts

```diff
@@ -134,10 +159,25 @@ export async function executePlugin(
  */
 export async function executePlugins(
   plugins: PluginConfig[],
+  options?: { progress: boolean },
 ): Promise<PluginReport[]> {
-  return await plugins.reduce(async (acc, pluginCfg) => {
+  const { progress = false } = options || {};
+
+  const progressName = 'Run Plugins';
+  const progressBar = progress ? getProgressBar(progressName) : MOCK_PROGRESS;
+
+  const pluginsResult = await plugins.reduce(async (acc, pluginCfg) => {
     const outputs = await acc;
+
+    progressBar.updateTitle(`Executing  ${chalk.bold(pluginCfg.title)}`);
+
     const pluginReport = await executePlugin(pluginCfg);
+    progressBar.incrementInSteps(plugins.length);
+
     return outputs.concat(pluginReport);
   }, Promise.resolve([] as PluginReport[]));
+
+  progressBar.endProgress('Done running plugins');
```

Comment:
An alternative to my previous comment - we don't necessarily need to stubbed progress bar at all if we use optional chaining in the method calls :thinking: 

```suggestion
  const progressBar = progress ? getProgressBar(progressName) : null;

  const pluginsResult = await plugins.reduce(async (acc, pluginCfg) => {
    const outputs = await acc;

    progressBar?.updateTitle(`Executing  ${chalk.bold(pluginCfg.title)}`);

    const pluginReport = await executePlugin(pluginCfg);
    progressBar?.incrementInSteps(plugins.length);

    return outputs.concat(pluginReport);
  }, Promise.resolve([] as PluginReport[]));

  progressBar?.endProgress('Done running plugins');
```

Reply:
I wanted to implement the „Null Object Pattern“ therefore I went with a stub